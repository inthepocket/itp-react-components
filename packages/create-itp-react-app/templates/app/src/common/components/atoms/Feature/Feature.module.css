.Feature {
  display: inline-block;
  margin-right: 5px;
  color: var(--colorSecondary900);
}

.Feature:hover {
  text-decoration: none;
}

.Feature::after {
  content: ",";
}

.Feature:last-child::after {
  display: none;
  content: "";
}
