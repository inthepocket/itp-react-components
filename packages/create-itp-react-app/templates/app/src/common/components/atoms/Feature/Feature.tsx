import * as React from 'react';
import { default as styles } from './Feature.module.css';

const Feature = ({ url, title }: FeatureProps) => {
  if (url) {
    return (
      <a className={styles.Feature} href={url} target="_blank">{title}</a>
    );
  }
  return (
    <span className={styles.Feature}>{title}</span>
  );
};

interface FeatureProps {
  url?: string;
  title: string;
}

export default Feature;
