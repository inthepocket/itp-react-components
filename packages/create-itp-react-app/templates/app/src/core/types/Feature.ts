export interface FeatureInterface {
  id: string;
  title: string;
  url?: string;
}

export interface FeaturesEntitiesInterface {
  [id: string]: FeatureInterface;
}

export interface FeaturesUIInterface {
  error?: any;
  isFetchingFeatures: boolean;
  items: string[];
}

export interface FeaturesPayloadInterface {
  entities: { features: FeaturesEntitiesInterface; };
  result: string[];
}

export interface FeaturesStateInterface {
  readonly entities?: FeaturesEntitiesInterface;
  readonly ui?: FeaturesUIInterface;
}
