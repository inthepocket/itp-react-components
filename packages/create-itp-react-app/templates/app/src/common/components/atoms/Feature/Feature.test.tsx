import * as React from 'react';
import reactTestRenderer from 'react-test-renderer';
import Feature from './Feature';

test('Feature snapshot: default', () => {
  const tree = reactTestRenderer
    .create((
      <Feature title="foo:bar" />
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Feature snapshot: with url', () => {
  const tree = reactTestRenderer
    .create((
      <Feature title="foo:bar" url="https://foo.bar/" />
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
