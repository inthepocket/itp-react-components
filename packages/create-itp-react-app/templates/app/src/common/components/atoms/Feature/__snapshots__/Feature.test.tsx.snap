// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature snapshot: default 1`] = `
<span
  className="Feature"
>
  foo:bar
</span>
`;

exports[`Feature snapshot: with url 1`] = `
<a
  className="Feature"
  href="https://foo.bar/"
  target="_blank"
>
  foo:bar
</a>
`;
